xm-commons-tenant/src/main/java/com/icthh/xm/commons/tenant/internal/TenantContextDataHolder.java

```diff
@@ -28,6 +28,10 @@ final class TenantContextDataHolder {
     private static ThreadLocal<Stack<TenantContextDataHolder>> parentHolderInstanceStack =
         ThreadLocal.withInitial(Stack::new);
 
+    static {
+        ThreadLocalAccessorRegistryInterceptor.register("holderInstance", holderInstance);
+    }
```

Comment:
why this changes is requeired?

Reply:
This changes was added to propagate `ThreadLocal<TenantContextDataHolder> holderInstance` between threads in reactive projects. `ThreadLocalAccessorRegistryInterceptor` registers threadLocal in micrometer context registry.

If the thread is registered, the application only needs to enable mdc context propagation by `Hooks.enableAutomaticContextPropagation();`

You could check this related conversation also: [https://github.com/xm-online/xm-commons/pull/245#discussion_r1472843375]